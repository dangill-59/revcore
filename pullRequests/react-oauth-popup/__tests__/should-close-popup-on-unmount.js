import OauthPopup from '../src/index.js';
import React from 'react';
import renderer from 'react-test-renderer';

test('OauthPopup closes popup on unmount', () => {
  const close = jest.fn();
  global.open = jest.fn().mockReturnValue({
    location: 'http://www.google.com',
    close,
  });
  const component = renderer.create(
    <OauthPopup>
      <div>coooooooool</div>
      <h2>siiiiiiiick</h2>
      <h2>leeegiiiiit</h2>
      <button>Doooooope</button>
    </OauthPopup>,
  );
  let popup = component.toJSON();
  expect(popup).toMatchSnapshot();
  popup.props.onClick();
  popup = component.toJSON();
  expect(global.open.mock.calls.length).toBe(1);
  component.unmount();
  expect(close.mock.calls.length).toBe(1);
  expect(global.open.mock.calls.length).toBe(1);
});
