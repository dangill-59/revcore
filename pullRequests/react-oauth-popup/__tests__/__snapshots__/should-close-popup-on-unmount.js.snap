// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OauthPopup closes popup on unmount 1`] = `
<div
  onClick={[Function]}
>
   
  <div>
    coooooooool
  </div>
  <h2>
    siiiiiiiick
  </h2>
  <h2>
    leeegiiiiit
  </h2>
  <button>
    Doooooope
  </button>
   
</div>
`;
